python/ray/util/collective/collective_group/nccl_collective_group.py

```diff
@@ -422,10 +431,51 @@ def _get_cuda_stream():
         # TODO: implement a simple stream manager.
         return cupy.cuda.Stream.null
 
-    # Note(Hao): too many bipolate code -- make some abstraction.
-    # def _collective_call(self, *args):
-    #     """Private method to encapsulate all collective calls"""
-    #     pass
+    def _collective(self, input_tensor, output_tensor, collective_fn,
+                    preprocess_fn, postprocess_fn):
```

Comment:
can you make these named arguments and default preprocess_fn and postprocess_fn to None?

Then, you don't need to drop the lambda everywhere

Reply:
sounds perfect!